src/Microsoft.DotNet.Build.CloudTestTasks/UploadToAzure.cs

```diff
@@ -101,42 +103,55 @@ public override bool Execute()
                         "GET",
                         dt,
                         req));
-                    
-                    Log.LogMessage(MessageImportance.Normal, "Sending request to check whether Container blobs exist");
-                    XmlDocument doc;
-                    using (HttpResponseMessage response = await client.SendAsync(req, ct))
-                    {
-                        doc = new XmlDocument();
-                        doc.LoadXml(await response.Content.ReadAsStringAsync());
-                    }
 
-                    XmlNodeList nodes = doc.DocumentElement.GetElementsByTagName("Blob");
-
-                    foreach (XmlNode node in nodes)
+                    Log.LogMessage(MessageImportance.Low, "Sending request to check whether Container blobs exist");
+                    using (HttpResponseMessage response = await client.SendAsync(req, ct))
                     {
-                        blobsPresent.Add(node["Name"].InnerText);
+                        if (response.IsSuccessStatusCode)
+                        {
+                            var doc = new XmlDocument();
+                            doc.LoadXml(await response.Content.ReadAsStringAsync());
+
+                            XmlNodeList nodes = doc.DocumentElement.GetElementsByTagName("Blob");
+
+                            foreach (XmlNode node in nodes)
+                            {
+                                blobsPresent.Add(node["Name"].InnerText);
+                            }
+
+                            Log.LogMessage(MessageImportance.Low, "Received response to check whether Container blobs exist");
+                            gotExistingBlobs = true;
```

Comment:
poorly worded.  Since you just logged "Sending request to check whether Container blobs exist", maybe "found container blobs" would be much more clear.